src/utils.js

```diff
@@ -287,6 +303,40 @@ genotet.utils.keySet = function(collection) {
 };
 
 /**
+ * Validates email address.
+ * @param {string} email
+ * @return {boolean}
+ */
+genotet.utils.validateEmail = function(email) {
```

Comment:
I think keeping all the validation functions setting in utils is better than writing one validateRegex function but calling them with different regex constants in different places. For example, if I want to validate username, calling validateUsername(username) is better than validateRegex(username, validate_username_const). Because the validation is called in some places, the caller only care about the username input, but not what the validate_username_const is. I think validateUsername(username) is a better interface.